### ISSUE TITLE
Code style (questions?)

### ISSUE THREAD

**Issue by dertuxmalwieder** [fcorbelli/zpaqfranz#20]
From the source code:

> From branch 51 all source code merged in one zpaqfranz.cpp
> aiming to make it as easy as possible to compile on "strange" systems (NAS, vSphere etc).

Does that really help? (Because it makes reading the code much harder.)

> The source is composed of the fusion of different software 
> from different authors, therefore there is no uniform style of programming. 

Wouldn't it make sense to run a tool [like astyle](http://astyle.sourceforge.net) over it to have a coherent *indentation* style, at least? (That would not violate any licenses.)

**Comment by fcorbelli** [comment]
> From the source code:
> 
> > From branch 51 all source code merged in one zpaqfranz.cpp
> > aiming to make it as easy as possible to compile on "strange" systems (NAS, vSphere etc).
> 
> Does that really help? (Because it makes reading the code much harder.)
> 

Yes, it really help fast deploying.  
There is no need for make, makefile, cmake etc. 
You don't have to download a compressed file, expand the source (how?), than "do" something.  
You build the VM, install gcc/clang (if not already here), and it's enough to download (in my case from my ftp server) a single file (via ...ftp command),compile it directly with an easy to remember command (if you do not have the possibility to make a copy-paste, for example for the console of a virtual machine)   

Reading the source code is almost useless (!), it contains a huge amount of very subtle optimizations developed over the course of about 5 years from Mahoney.  
Certainly a person as brilliant as "messy" in the way of programming. Or at least, with a different style from mine  (if it's simple, MAYBE it will works).

Unfortunately, at that time, I did not know zpaq and did not speak with its author.
in the development phase.  
When I proposed the first changes, the "core" had already been done, and the forum users (who had contributed at the time) seem to have disappeared  
Today he is retired, and basically does not answer questions (at least, not mine)  
Rebuilding "how it works is", for me, a work in progress

> > The source is composed of the fusion of different software
> > from different authors, therefore there is no uniform style of programming.
> 
> Wouldn't it make sense to run a tool [like astyle](http://astyle.sourceforge.net) over it to have a coherent _indentation_ style, at least? (That would not violate any licenses.)

There is code from authors who write practically like me, and others with a completely different style.   
Refactoring would take so much time, and effort - there are very complex "pieces" of software - that it doesn't make sense for a free project.  
At least, in my opinion

Why no uniform indentation?  
Because I can "scope" faster "my" code, or reworked by me, against "untouched"

This is NOT mine
```
  // Read checksum
  int c=dec.in->get();
  if (c==254) {
    if (sha1string) sha1string[0]=0;  // no checksum
  }
  else if (c==253) {
    if (sha1string) sha1string[0]=1;
    for (int i=1; i<=20; ++i) {
      c=dec.in->get();
      if (sha1string) sha1string[i]=c;
    }
  }
  else
    unzerror("missing end of segment marker");
```
